includes/html/modal/device_maintenance.inc.php

```diff
@@ -0,0 +1,124 @@
+ * @copyright  2020 Thomas Berberich
+ * @author     Thomas Berberich <sourcehhdoctor@gmail.com>
+ */
+
+use LibreNMS\Alerting\QueryBuilderParser;
+
+if (!Auth::user()->hasGlobalAdmin()) {
+    die('ERROR: You need to be admin');
+}
+
+$hour_steps = range(0, 23, 1);
+$minute_steps = [0, 30];
+$exclude_durations = ['0:00'];
+
+$maintenance_duration_list = [];
+foreach ($hour_steps as $hour) {
+    foreach ($minute_steps as $min) {
+        if (empty($hour) && empty($min)) {
+            continue;
+        }
+        $str_hour = $hour;
+        $str_min = $min < 10 ? '0'.$min : $min;
+
+        $duration = $str_hour.':'.$str_min;
+
+        if (in_array($duration, $exclude_durations)) {
+            continue;
+        }
+        $maintenance_duration_list[] = $duration;
+    }
+}
+?>
+<div class="modal fade" id="device_maintenance_modal" tabindex="-1" role="dialog" aria-labelledby="device_edit" aria-hidden="true">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                <h5 class="modal-title" id="search_alert_rule_list">Device Maintenance</h5>
+            </div>
+            <div class="modal-body">
+                <form method="post" role="form" id="sched-form" class="form-horizontal schedule-maintenance-form">
+                    <?php echo csrf_field() ?>
+                    <div class="form-group">
+                        <label for="notes" class="col-sm-4 control-label">Notes: </label>
+                        <div class="col-sm-8">
+                            <textarea class="form-control" id="notes" name="notes" placeholder="Maintenance notes"></textarea>
+                        </div>
+                    </div>
+                    <div class="form-group">
+                        <label for="duration" class="col-sm-4 control-label">Duration: </label>
+                        <div class="col-sm-8">
+                            <select name='duration' id='duration' class='form-control input-sm'>
+                                <?php foreach ($maintenance_duration_list as $dur) { ?>
+                                <option value='<?=$dur?>'><?=$dur?>h</option>
+                                <?php }?>
+                            </select>
+                        </div>
+                    </div>
+                    <div class="form-group">
+                        <label for="maintenance-submit" class="col-sm-4 control-label"></label>
+                        <div class="col-sm-8">
+                            <button type="submit" id="maintenance-submit" data-device_id="<?php echo($device['device_id']); ?>" <?php echo(\LibreNMS\Alert\AlertUtil::isMaintenance($device['device_id']) ? 'disabled class="btn btn-warning"' : 'class="btn btn-success"')?> name="maintenance-submit"><i class="fa fa-wrench"></i> Start Maintenance</button>
```

Comment:
This button should not really have the wrench.  That indicates maintenance active on the previous button.
Also, why let them launch the modal when you are just going to disable the save button if there is already a maintenance window active?  What about extending maintenance windows?

Reply:
wrench removed.